frontend/src/components/dashboard.tsx

```diff
@@ -12,7 +12,10 @@ import { Settings } from "./settings";
 import { ConnectedSidebar } from "./sidebar";
 import { SidebarBurger } from "./sidebar-burger";
 
-interface ISidebarAwareProps {
+/**
+ * Interface for whether or not the sidebar is being shown.
+ */
+export interface ISidebarAwareProps {
```

Comment:
I'm not a fan of sharing props definitions across different modules, but it should be fine here.

Reply:
Only did it that way so that I could effortlessly move the headerbar into a seperate file, as I think putting it into the same file would just clutter the code / the file.